shared/utils/rootItemsGetter.ts

```diff
@@ -0,0 +1,86 @@
+import { ContentItem } from 'kentico-cloud-delivery';
+import { IWebhookContentItem } from '../external/models';
+
+interface IContext {
+    readonly visitedItems: string[],
+    readonly rootItemCodenames: string[],
+    readonly newItemsToVisit: ContentItem[],
+    readonly allItems: ContentItem[],
+}
+
+export const getRootCodenamesOfSingleItem = (item: IWebhookContentItem, allItems: ContentItem[]): string[] => {
+    if (item.type === 'zapi_specification') {
```

Comment:
Please,  extract the name root element type into a creator function:
```javascript
export const getRootCodenamesOfSingleItem = (rootElementType: string) => (item: IWebhookContentItem, allItems: ContentItem[]): string[] => {
    if (item.type === rootElementType) {
...
```
This will make the  algoritm generic (for `ContentItem`s)...